server/static/static.go

```diff
@@ -120,10 +131,33 @@ func serveIndexTemplate(env environment.Env, tpl *template.Template, version str
 	}
 	err := tpl.ExecuteTemplate(w, indexTemplateFilename, &cfgpb.FrontendTemplateData{
 		Config:           &config,
-		JsEntryPointPath: *jsEntryPointPath,
+		JsEntryPointPath: jsPath,
 		GaEnabled:        !*disableGA,
 	})
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 }
+
+func AppBundleHash(bundleFS fs.FS) string {
+	var hashBytes []byte
+	if data, err := fs.ReadFile(bundleFS, "sha.sum"); err == nil {
+		hashBytes = data
+	}
+	if hashBytes != nil {
+		return strings.TrimSpace(string(hashBytes))
+	}
+	return ""
```

Comment:
nit: log the error if it is non-nil (and reading the data outside the `if` scope simplifies things a bit):

```go
data, err := fs.ReadFile(bundleFS, "sha.sum")
if err != nil {
  log.Errorf("Failed to read app checksum: %s", err)
  return ""
}
return strings.TrimSpace(string(hashBytes))
```

Reply:
Made this a real error.